main.tf

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  default_ack_deadline_seconds = 10
+}
+
+resource "google_pubsub_topic" "topic" {
+  name    = "${var.topic}"
+  project = "${var.project_id}"
+}
+
+resource "google_pubsub_subscription" "push_subscriptions" {
+  count                = "${length(var.push_subscriptions)}"
+  name                 = "${lookup(var.push_subscriptions[count.index], "name")}"
+  topic                = "${var.topic}"
+  project              = "${var.project_id}"
+  ack_deadline_seconds = "${lookup(var.push_subscriptions[count.index], "ack_deadline_seconds", local.default_ack_deadline_seconds)}"
+
+  push_config {
+    push_endpoint = "${lookup(var.push_subscriptions[count.index], "push_endpoint")}"
```

Comment:
terraform doesn't support nested map, so I think we should implement the `x-goog-version` explicity as a temporary, instead of implementing mapping to attributes.

Reply:
Agreed.

Reply:
Fixed via: https://github.com/terraform-google-modules/terraform-google-pubsub/pull/1/commits/067b503014d7ca3a97a84e8aaa7089416cb01b42

Reply:
To be clear, I don't think this should be hard-coded.

Instead I was imaging there would be a `x-goog-version` argument on the list which would be passed through.

Reply:
Ah ok. I've tried to make sure the interface for the case but it's not good.
How about this? https://github.com/terraform-google-modules/terraform-google-pubsub/pull/1/commits/039a43c453f20b398ff64949d64ad03d447431e3

Reply:
LGTM, nice job!